x-pack/plugins/security_solution/public/management/components/endpoint_responder/lib/console_commands_definition.ts

```diff
@@ -523,6 +524,74 @@ export const getEndpointConsoleCommands = ({
       privileges: endpointPrivileges,
     }),
   });
+  if (crowdstrikeRunScriptEnabled) {
+    consoleCommands.push({
+      name: 'runscript',
+      about: getCommandAboutInfo({
+        aboutInfo: CROWDSTRIKE_CONSOLE_COMMANDS.runscript.about,
+        isSupported: doesEndpointSupportCommand('runscript'),
+      }),
+      RenderComponent: () => null,
+      meta: {
+        agentType,
+        endpointId: endpointAgentId,
+        capabilities: endpointCapabilities,
+        privileges: endpointPrivileges,
+      },
+      exampleUsage: `C:\\> runscript -CloudFile="CloudScript1.ps1" -CommandLine="-Verbose true"
+Run a script saved to the CrowdStrike cloud with the specified command line arguments
+C:\\> runscript -CloudFile="CloudScript1.ps1" -CommandLine="-Verbose true" -Timeout=180
+Run a script saved to the CrowdStrike cloud with the specified command line arguments and 180 seconds timeout
+C:\\> runscript -Raw=\`\`\`Get-ChildItem .\`\`\` -CommandLine=""
+Run a raw script whose entire contents are provided in the "-Raw=" flag
+C:\\> runscript -HostPath="C:\\temp\\LocalScript.ps1" -CommandLine="-Verbose true"
+Run a script from a path on the remote host with the specified command line arguments`,
+      exampleInstruction: CROWDSTRIKE_CONSOLE_COMMANDS.runscript.about,
+      validate: capabilitiesAndPrivilegesValidator(agentType),
+      mustHaveArgs: true,
+      args: {
+        Raw: {
+          required: false,
+          allowMultiples: false,
+          about: CROWDSTRIKE_CONSOLE_COMMANDS.runscript.args.raw.about,
+          mustHaveValue: 'non-empty-string',
+        },
+        CloudFile: {
+          required: false,
+          allowMultiples: false,
+          about: CROWDSTRIKE_CONSOLE_COMMANDS.runscript.args.cloudFile.about,
+          mustHaveValue: 'non-empty-string',
+        },
+        CommandLine: {
+          required: false,
+          allowMultiples: false,
+          about: CROWDSTRIKE_CONSOLE_COMMANDS.runscript.args.commandLine.about,
+          mustHaveValue: 'non-empty-string',
+        },
+        HostPath: {
+          required: false,
+          allowMultiples: false,
+          about: CROWDSTRIKE_CONSOLE_COMMANDS.runscript.args.hostPath.about,
+          mustHaveValue: 'non-empty-string',
+        },
+        Timeout: {
+          required: false,
+          allowMultiples: false,
+          about: CROWDSTRIKE_CONSOLE_COMMANDS.runscript.args.timeout.about,
+          mustHaveValue: 'non-empty-string',
+        },
```

Comment:
I'm a bit confused on this setup for the arguments... Should at least one of these be required?  if so, then all arguments configuration should have `exclusiveOr: true` set on them.

Reply:
Good point, I wasn't aware `exclusiveOr:true` existed - will adjust the code 👍 Thanks!

Reply:
Reading through the options again - can you detail in the issue what the input is to these arguments and how they should be used?

Example: 

- How will a user know what `--CloudFile` should be set to? Can this be "anything"? or does it have to match a file in the Crowdstrike system?  if it has to match a file in the Crowdstrike system, do we have an API in crowdstrike that can pull that data in? because if so, then we should provide a better UI in the console for it (a popup to pick it)
- Is `CommandLine` an argument that can be used with all other arguments (in combination with)? or is it only applicable to some (ex. maybe only applicable to `CloudFile` and `HostPath`?)

Reply:
Also: I played with CS Console and it's possible to pass multiple arguments eg. 
`runscript --CloudFile='test1' --Raw='ls' --HostPath='test1' `. 
However no matter in which order I've put the commands -it always started with HostPath, then CloudFile and Raw as last - strange behavior. 
Anyway - I think we should mark all the arguments as optional. It's how it's done in `CrowdStrike`, and if arguments are missing - API will return an error eg.
 `stderr:"runscript: Either -Raw or -HostPath must be specified\n"`
 
 What do you think?